realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -0,0 +1,604 @@
+        return lastImpl(false, defaultValue);
+
+    }
+
+    private E lastImpl(boolean shouldThrow, E defaultValue) {
+        UncheckedRow row = collection.lastUncheckedRow();
+
+        if (row != null) {
+            return realm.get(classSpec, className, row);
+        } else {
+            if (shouldThrow) {
+                throw new IndexOutOfBoundsException("No results were found.");
+            } else {
+                return defaultValue;
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void deleteFromRealm(int location) {
+        // TODO: Implement the deleteLast in OS level and do check there!
+        realm.checkIfValidAndInTransaction();
+        collection.delete(location);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean deleteAllFromRealm() {
+        realm.checkIfValid();
+        if (size() > 0) {
+            collection.clear();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns an iterator for the results of a query. Any change to Realm while iterating will cause this iterator to
+     * throw a {@link ConcurrentModificationException} if accessed.
+     *
+     * @return an iterator on the elements of this list.
+     * @see Iterator
+     */
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public Iterator<E> iterator() {
+        return new RealmCollectionIterator();
+    }
+
+    /**
+     * Returns a list iterator for the results of a query. Any change to Realm while iterating will cause the iterator
+     * to throw a {@link ConcurrentModificationException} if accessed.
+     *
+     * @return a ListIterator on the elements of this list.
+     * @see ListIterator
+     */
+    @Override
+    public ListIterator<E> listIterator() {
+        return new RealmCollectionListIterator(0);
+    }
+
+    /**
+     * Returns a list iterator on the results of a query. Any change to Realm while iterating will cause the iterator to
+     * throw a {@link ConcurrentModificationException} if accessed.
+     *
+     * @param location the index at which to start the iteration.
+     * @return a ListIterator on the elements of this list.
+     * @throws IndexOutOfBoundsException if {@code location < 0 || location > size()}.
+     * @see ListIterator
+     */
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public ListIterator<E> listIterator(int location) {
+        return new RealmCollectionListIterator(location);
+    }
+
+    // Sorting
+
+    // aux. method used by sort methods
+    private long getColumnIndexForSort(String fieldName) {
+        if (fieldName == null || fieldName.isEmpty()) {
+            throw new IllegalArgumentException("Non-empty field name required.");
+        }
+        if (fieldName.contains(".")) {
+            throw new IllegalArgumentException("Sorting using child object fields is not supported: " + fieldName);
```

Comment:
This is for the aggregates used in the class. I changed the exception message in this PR and created an issue https://github.com/realm/realm-java/issues/4194